assets/src/edit-story/components/richText/karma/inlineSelection.js

```diff
@@ -0,0 +1,366 @@
+        expect(bold.checked).toBe(true);
+        expect(fontWeight.value).toBe('(multiple)');
+
+        // Toggle it by pressing the bold button
+        await data.fixture.events.click(bold.button);
+
+        // Verify bold is now off and font weight is Regular
+        expect(bold.checked).toBe(false);
+        expect(fontWeight.value).toBe('Regular');
+
+        // Exit edit-mode
+        await data.fixture.events.keyboard.press('Escape');
+
+        // Assume text content to now be formatting-free
+        const actual = getTextContent();
+        const expected = `Fill in some text`;
+        expect(actual).toBe(expected);
+      });
+
+      it('should make bold+regular selection bold when toggling', async () => {
+        const {
+          bold,
+          fontWeight,
+        } = data.fixture.editor.inspector.designPanel.textStyle;
+
+        // Select second and third characters (700 and 400)
+        await setSelection(1, 3);
+
+        // Check that bold toggle is off but font weight is "multiple"
+        expect(bold.checked).toBe(false);
+        expect(fontWeight.value).toBe('(multiple)');
+
+        // Toggle it by pressing the bold button
+        await data.fixture.events.click(bold.button);
+
+        // Verify bold is now on and font weight is Bold
+        expect(bold.checked).toBe(true);
+        expect(fontWeight.value).toBe('Bold');
+
+        // Exit edit-mode
+        await data.fixture.events.keyboard.press('Escape');
+
+        // Assume text content to be correctly formatted
+        const actual = getTextContent();
+        const expected = `<span style="font-weight: 900">F</span><span style="font-weight: 700">il</span>l in some text`;
+        expect(actual).toBe(expected);
+      });
+
+      it('should make black+bold+regular selection black when toggling', async () => {
+        const {
+          bold,
+          fontWeight,
+        } = data.fixture.editor.inspector.designPanel.textStyle;
+
+        // Select first three characters (900, 700 and 400)
+        await setSelection(0, 3);
+
+        // Check that bold toggle is off but font weight is "multiple"
+        expect(bold.checked).toBe(false);
+        expect(fontWeight.value).toBe('(multiple)');
+
+        // Toggle it by pressing the bold button
+        await data.fixture.events.click(bold.button);
+
+        // Verify bold is now on and font weight is Black
+        expect(bold.checked).toBe(true);
+        expect(fontWeight.value).toBe('Black');
+
+        // Exit edit-mode
+        await data.fixture.events.keyboard.press('Escape');
+
+        // Assume text content to be correctly formatted
+        const actual = getTextContent();
+        const expected = `<span style="font-weight: 900">Fil</span>l in some text`;
+        expect(actual).toBe(expected);
+      });
+    });
+
+    it('should apply global formats (here line height) even when a selection is present', async () => {
+      // Expect line height to be default "Roboto"
```

Comment:
Not sure I understand this comment -- is this correct?

Reply:
Comment removed in c9da0ae2f.